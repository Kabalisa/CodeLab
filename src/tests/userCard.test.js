import UserCard from '../components/userCard';
describe('UserCard component test', () => {
  const props = {
    navigation: { navigate: jest.fn() },
    item: { node: { avatarUrl: 'avatarUrl', login: 'name' } }
  };
  it('renders without error', () => {
    const wrapper = shallow(<UserCard {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('press on the image', () => {
    const wrapper = shallow(<UserCard {...props} />);
    const imageButton = wrapper.find('[id="image"]');
    imageButton.props().onPress();
  });
  it('press on the name', () => {
    const wrapper = shallow(<UserCard {...props} />);
    const nameButton = wrapper.find('[id="name"]');
    nameButton.props().onPress();
  });
});
