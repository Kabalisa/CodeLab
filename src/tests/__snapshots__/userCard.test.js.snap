// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard component test renders without error 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "marginTop": 20,
    }
  }
>
  <TouchableOpacity
    activeOpacity={0.2}
    id="image"
    onPress={[Function]}
  >
    <Image
      source={
        Object {
          "uri": "avatarUrl",
        }
      }
      style={
        Object {
          "borderRadius": 35,
          "height": 70,
          "width": 70,
        }
      }
    />
  </TouchableOpacity>
  <TouchableOpacity
    activeOpacity={0.2}
    id="name"
    onPress={[Function]}
    style={
      Object {
        "alignSelf": "flex-end",
        "bottom": 5,
        "left": 30,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 18,
          "fontWeight": "bold",
        }
      }
    >
      name
    </Text>
  </TouchableOpacity>
</View>
`;
